moto/kafka/models.py

```diff
@@ -0,0 +1,373 @@
+from moto.core.common_models import BaseModel
+from moto.utilities.utils import get_partition
+
+from ..utilities.tagging_service import TaggingService
+
+
+class FakeKafkaCluster(BaseModel):
+    def __init__(
+        self,
+        cluster_name: str,
+        account_id: str,
+        region_name: str,
+        cluster_type: str,
+        tags: Optional[Dict[str, str]] = None,
+        broker_node_group_info: Optional[Dict[str, Any]] = None,
+        kafka_version: Optional[str] = None,
+        number_of_broker_nodes: Optional[int] = None,
+        configuration_info: Optional[Dict[str, Any]] = None,
+        serverless_config: Optional[Dict[str, Any]] = None,
+        encryption_info: Optional[Dict[str, Any]] = None,
+        enhanced_monitoring: str = "DEFAULT",
+        open_monitoring: Optional[Dict[str, Any]] = None,
+        logging_info: Optional[Dict[str, Any]] = None,
+        storage_mode: str = "LOCAL",
+        current_version: str = "1.0",
+        client_authentication: Optional[Dict[str, Any]] = None,
+        state: str = "CREATING",
+        active_operation_arn: Optional[str] = None,
+        zookeeper_connect_string: Optional[str] = None,
+        zookeeper_connect_string_tls: Optional[str] = None,
+    ):
+        # General attributes
+        self.cluster_id = str(uuid.uuid4())
+        self.cluster_name = cluster_name
+        self.account_id = account_id
+        self.region_name = region_name
+        self.cluster_type = cluster_type
+        self.tags = tags or {}
+        self.state = state
+        self.creation_time = datetime.now().isoformat()
+        self.current_version = current_version
+        self.active_operation_arn = active_operation_arn
+        self.arn = self._generate_arn()
+
+        # Attributes specific to PROVISIONED clusters
+        self.broker_node_group_info = broker_node_group_info
+        self.kafka_version = kafka_version
+        self.number_of_broker_nodes = number_of_broker_nodes
+        self.configuration_info = configuration_info
+        self.encryption_info = encryption_info
+        self.enhanced_monitoring = enhanced_monitoring
+        self.open_monitoring = open_monitoring
+        self.logging_info = logging_info
+        self.storage_mode = storage_mode
+        self.client_authentication = client_authentication
+        self.zookeeper_connect_string = zookeeper_connect_string
+        self.zookeeper_connect_string_tls = zookeeper_connect_string_tls
+
+        # Attributes specific to SERVERLESS clusters
+        self.serverless_config = serverless_config
+
+    def _generate_arn(self) -> str:
+        resource_type = (
+            "cluster" if self.cluster_type == "PROVISIONED" else "serverless-cluster"
+        )
+        partition = get_partition(self.region_name)
+        return f"arn:{partition}:kafka:{self.region_name}:{self.account_id}:{resource_type}/{self.cluster_id}"
+
+    def to_dict(self) -> Dict[str, Any]:
```

Comment:
Can we not remove this method altogether @jamarcelin? As far as I can tell, it's not used anywhere